tests/host_tools/memory.py

```diff
@@ -11,20 +11,32 @@
 MEMORY_COP_TIMEOUT = 1
 
 
-def threaded_memory_monitor(mem_size_mib, pid):
+class MemoryUsageExceededException(Exception):
+    """A custom exception containing details on excessive memory usage."""
+
+    def __init__(self, usage):
+        """Compose the error message containing the memory consumption."""
+        super(MemoryUsageExceededException, self).__init__(
+            'Memory usage exceeded maximum threshold: {} MiB.\n'
```

Comment:
nit: From this phrase it seems you are actually printing the threshold and not the violator value.